crates/oxc_linter/src/rules/no_setter_return.rs

```diff
@@ -0,0 +1,309 @@
+        ("Object.create(foo, { bar: { set: (val) => { (val) => 1; } } })", None),
+        ("Object.defineProperty(foo, 'bar', 'baz', { set(val) { return 1; } })", None),
+        ("Object.defineProperty(foo, { set(val) { return 1; } }, 'bar')", None),
+        ("Object.defineProperty({ set(val) { return 1; } }, foo, 'bar')", None),
+        ("Reflect.defineProperty(foo, 'bar', 'baz', { set(val) { return 1; } })", None),
+        ("Reflect.defineProperty(foo, { set(val) { return 1; } }, 'bar')", None),
+        ("Reflect.defineProperty({ set(val) { return 1; } }, foo, 'bar')", None),
+        ("Object.defineProperties(foo, bar, { baz: { set(val) { return 1; } } })", None),
+        ("Object.defineProperties({ bar: { set(val) { return 1; } } }, foo)", None),
+        ("Object.create(foo, bar, { baz: { set(val) { return 1; } } })", None),
+        ("Object.create({ bar: { set(val) { return 1; } } }, foo)", None),
+        ("Object.DefineProperty(foo, 'bar', { set(val) { return 1; } })", None),
+        ("Reflect.DefineProperty(foo, 'bar', { set(val) { if (val) { return 1; } } })", None),
+        (
+            "Object.DefineProperties(foo, { bar: { set(val) { try { return 1; } catch(e){} } } })",
+            None,
+        ),
+        ("Object.Create(foo, { bar: { set: function(val) { return 1; } } })", None),
+        ("object.defineProperty(foo, 'bar', { set(val) { return 1; } })", None),
+        ("reflect.defineProperty(foo, 'bar', { set(val) { if (val) { return 1; } } })", None),
+        (
+            "Reflect.defineProperties(foo, { bar: { set(val) { try { return 1; } catch(e){} } } })",
+            None,
+        ),
+        ("object.create(foo, { bar: { set: function(val) { return 1; } } })", None),
+        ("Reflect.defineProperty(foo, 'bar', { set(val) { if (val) { return 1; } } })", None),
+        (
+            "/* globals Object:off */ Object.defineProperty(foo, 'bar', { set(val) { return 1; } })",
+            None,
+        ),
+        (
+            "Object.defineProperties(foo, { bar: { set(val) { try { return 1; } catch(e){} } } })",
+            None,
+        ),
+        ("let Object; Object.defineProperty(foo, 'bar', { set(val) { return 1; } })", None),
+        (
+            "function f() { Reflect.defineProperty(foo, 'bar', { set(val) { if (val) { return 1; } } }); var Reflect;}",
+            None,
+        ),
+        (
+            "function f(Object) { Object.defineProperties(foo, { bar: { set(val) { try { return 1; } catch(e){} } } }) }",
+            None,
+        ),
+        (
+            "if (x) { const Object = getObject(); Object.create(foo, { bar: { set: function(val) { return 1; } } }) }",
+            None,
+        ),
+        (
+            "x = function Object() { Object.defineProperty(foo, 'bar', { set(val) { return 1; } }) }",
+            None,
+        ),
+    ];
+
+    let fail = vec![
+        // ("({ set a(val){ return val + 1; } })", None),
```

Comment:
@Boshen Objects having setter functions does not get detected as MethodDefinition::Setter AstNode kind.